sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -100,3 +102,19 @@
 // clang emits "warning:" in the message pragma output
 #define __SYCL_WARNING(msg) __pragma(message(msg))
 #endif // __GNUC__
+
+// defined __SYCL_UNROLL to add pragma/attribute unroll to a loop
+#ifndef __SYCL_UNROLL
+#if defined(__clang__)
+#define __SYCL_UNROLL(x) _Pragma(STRINGIFY(unroll_count x))
+#elif defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER)
+#define __SYCL_UNROLL(x) _Pragma(STRINGIFY(unroll x))
```

Comment:
I think the order matters here because `__clang__` and `__INTEL_LLVM_COMPILER` are both defined at the same time. So I think we should check for the Intel compilers *first* and then fall back on Clang. WDYT?

Reply:
Indeed, applied